src/Compilers/Core/Portable/CommandLine/AnalyzerConfig.SectionNameMatching.cs

```diff
@@ -125,8 +126,15 @@ internal static bool TryUnescapeSectionName(string sectionName, out string? esca
                 {
                     sb.Append(lexer.EatCurrentCharacter());
                 }
+                else
```

Comment:
Given that we never expect this to fail, I think we should change it to `UnescapeGlobalSectionName()` and just `Debug.Assert()` that the `tokenKind` _is_ a `SimpleCharacter` and make it clear that its only useful for the global sections.

Reply:
Actually lets just move it to `GlobalAnalyzerConfig` altogether given that's the only place its used.

Reply:
@chsienki Moving this to GlobalAnalyzerConfig means making the lexer internal reather than private to this type. I can't tell if that's a feature or not.